### ISSUE TITLE
Add onCrashedLastRun

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-dotnet#735]
Add a callback onCrashedLastRun to SentryOptions that is called by the SDK passing the event
when Sentry is initialized and the last program execution terminated with a crash.

PR in Cocoa: getsentry/sentry-cocoa#808
Docs in Cocoa: https://docs.sentry.io/platforms/apple/enriching-events/user-feedback/
The issue on Java: https://github.com/getsentry/sentry-java/issues/1125

**Comment by dalux-era** [comment]
Hi. 
Could this feature get a prioritization bump? A lot of people are switching from AppCenter to Sentry. It would definitely help migrating for all the .NET projects that had this feature in AppCenter - Crashes.HasCrashedInLastSessionAsync().

**Comment by bruno-garcia** [comment]
@aritchie this could be added just for iOS and Android if that makes it easier, as it just binds to the native sdks. 

If you have a chance to peek at the code, any idea of the effort to add this? 


@vaind @bitsandfoxes did we get this in Unity to be a reference?

**Comment by aritchie** [comment]
@bruno-garcia I'm actually already working on this over at #3958 for iOS.  It already exists in part on Android.

**Comment by jamescrosswell** [comment]
> [@bruno-garcia](https://github.com/bruno-garcia) I'm actually already working on this over at [#3958](https://github.com/getsentry/sentry-dotnet/pull/3958) for iOS. It already exists in part on Android.

Just bumped into [this](https://github.com/dotnet/maui/discussions/653#discussioncomment-12116908) btw - which might be an interesting workaround if we don't have access to the file system. Although, I'm not sure what happens to performance if we try to store an envelope as a preference.

**Comment by vaind** [comment]
>  did we get this in Unity to be a reference?

Yeah, for example on iOS there's a bool function exposed so it's very simple to do: https://github.com/getsentry/sentry-unity/blob/0059c77876e3746d50998af5b9ca8c6adbff6f52/package-dev/Plugins/iOS/SentryNativeBridge.m#L49
